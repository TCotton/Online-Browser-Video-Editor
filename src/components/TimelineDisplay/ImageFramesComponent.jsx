import React from 'react';
import PropTypes from 'prop-types';
import ErrorBoundary from "../ErrorBoundary/Error";

export const ImageFramesComponent = (props) => {
const { images } = props
    return (
        images && Array.isArray(images) && images.length > 0? (
            <ErrorBoundary>
                {/* eslint-disable-next-line react/no-array-index-key */}
                {images.map((x,i) => <div className="image-container" key={i} data-testid="ImageFramesComponent"><img src={x} alt="" className="images"/></div>)}
            </ErrorBoundary>
        ) : null
    )
}

ImageFramesComponent.propTypes = {
    images: PropTypes.arrayOf(PropTypes.string).isRequired
};