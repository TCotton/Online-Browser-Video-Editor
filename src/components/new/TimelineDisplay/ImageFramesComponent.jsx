import React from 'react';
import {useSelector} from "react-redux";
import {imageArray} from '../slices/imageSlice';

export const ImageFramesComponent = () => {
   const imageAry = useSelector(imageArray);

    const images = imageAry.map((x) =>
        <img src={x} alt="" className={`images`} />
    );

    return (
        {images}
    )
}