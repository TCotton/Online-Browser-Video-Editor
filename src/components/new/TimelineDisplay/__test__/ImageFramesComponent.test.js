import React from "react";
import {render} from "@testing-library/react";
import {ImageFramesComponent} from '../ImageFramesComponent';

describe('ImageFramesComponent', () => {

    it('should render successfully', () => {
        const props = {
            images: ['image.png']
        }
        const {container} = render(<ImageFramesComponent {...props} />);
        expect(container.querySelector("[data-testid=\"ImageFramesComponent\"]")).not.toBeNull();
    });

    it('should match snapshot', () => {
        const props = {
            images: ['image.png']
        }
        const {baseElement} = render(<ImageFramesComponent {...props} />);
        expect(baseElement).toMatchSnapshot();
    });

    it('should contain image with src of `image.png`', () => {
        const props = {
            images: ['image.png']
        }
        const {container} = render(<ImageFramesComponent {...props} />);
        const image = container.querySelector("[data-testid=\"ImageFramesComponent\"] img");
        expect(image.src).toContain('image.png');
    });

});