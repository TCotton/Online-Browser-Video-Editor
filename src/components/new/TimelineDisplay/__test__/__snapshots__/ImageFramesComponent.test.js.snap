// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageFramesComponent should match snapshot 1`] = `
<body>
  <div>
    <div
      class="image-container"
      data-testid="ImageFramesComponent"
    >
      <img
        alt=""
        class="images"
        src="image.png"
      />
    </div>
  </div>
</body>
`;
